basics/fading/fading.toit

```diff
@@ -0,0 +1,33 @@
+/**
+Fading LED example code.
+
+This example increases and decreases the lighting level of a LED over and over again.
+
+How to wire the LED: 
+  - Place a 220ohm resistor in between the anode of the LED and GPIO19 and connect the cathode to ground.
+  - The resistor will increase the life of your LED.
+*/
+
+import gpio.pwm as gpio
+import gpio
+
+/// Defining the GPIO pin for the LED.
+LED ::= 19
+
+/// The configured range of the PWM.
+DUTY_MIN ::= 0.0
+DUTY_MAX ::= 1.0
+CHANGE_RATE ::= 0.01  // The amount of change every 5ms.
+
+main:
+  step := CHANGE_RATE
+  duty := DUTY_MIN
+  led_pwm ::= gpio.Pwm --frequency=100
+  led := gpio.Pin LED
+  fading_led := led_pwm.start led
+  while true:
+    fading_led.set_duty_factor(duty)
+    duty += step
+    if not DUTY_MIN <= duty <= DUTY_MAX:
+      step = -step
```

Comment:
My mistake. I forgot to bring it back into duty range.
```suggestion
      step = -step
      duty += step
```

This means that we stay at the min and max for 2 cycles. I think that's ok.